compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -384,15 +384,19 @@ impl<'tcx> AsmCodegenMethods<'tcx> for CodegenCx<'_, 'tcx> {
     ) {
         let asm_arch = self.tcx.sess.asm_arch.unwrap();
 
-        // Default to Intel syntax on x86
-        let intel_syntax = matches!(asm_arch, InlineAsmArch::X86 | InlineAsmArch::X86_64)
-            && !options.contains(InlineAsmOptions::ATT_SYNTAX);
-
         // Build the template string
         let mut template_str = String::new();
-        if intel_syntax {
-            template_str.push_str(".intel_syntax\n");
+
+        // Select the assembly syntax. On non-X86 platforms att is used by default, on X86 we use
```

Comment:
Slight correction to the comment: the name AT&T refers specifically to an x86 asm syntax. There is no such thing as AT&T syntax on non-x86 platforms.

Reply:
Huh, I've always thought of it as intel and "the other one". At least s390x, powerpc, etc seem more inspired by AT&T.

Anyway, I rephrased that comment.

Reply:
Intel syntax is closer to the standard asm syntax on most architectures with the destination register coming first. AT&T syntax is actually based on PDP-11 assembly since that was the original platform for UNIX and the x86 port adapted the existing assembler instead of using the standard syntax.